### DISCUSSION TITLE
Add another

### DISCUSSION THREAD

**Discussion by robertjmccarthy** [ministryofjustice/moj-frontend#686] [RELEASED: Components, pages and patterns]
The [add another component](https://design-patterns.service.justice.gov.uk/components/add-another/) is live in the MoJ Design System.

Use this discussion to suggest changes and improvements to this component.

**Comment by Whersi** [comment]
## Ticket number

https://github.com/ministryofjustice/moj-frontend/issues/551.

The details below is pulled from the ticket.

The [Add another](https://design-patterns.service.justice.gov.uk/components/add-another/) documentation and example are designed so that JavaScript is required. It mentions this:
> The add another component relies on JavaScript. When JavaScript is not available, your page should reload with the additional form elements if the "Add another" button is pressed.

In real services we require all functionality to work without JavaScript, so this is not sufficient and we should update the documentation to start from a server-side approach and add the JavaScript functionality as a progressive enhancement.

The main changes needed to the example itself are:
- Add a remove button with class `moj-add-another__remove-button` (JS doesn't add a new one if it already exists)
- Add suitable `name` attributes to buttons so that the server-side can hook in

We should then document the server-side logic required, and then introduce what adding the JS functionality does.

I've set up [an example project in Glitch](https://glitch.com/~mojds-add-another) to demo working server-side functionality.

## Ticket number

https://github.com/ministryofjustice/moj-frontend/issues/160

The details below is pulled from the ticket.

When you use [the Add another component](https://moj-design-system.herokuapp.com/components/add-another) to add an additional item, the legend on the form is identical to the original one: in the MOJDS docs example there would be two or more legends called "Person".

This makes it hard for users with screen readers to navigate the form, because the screen reader cannot distinguish between each fieldset.

To make this clearer, we should automatically enumerate the legends by appending an index number with JavaScript. i.e. when you "add another", the original legend will become "Person 1" and the new one will be "Person 2".

In order to support services which don't want this distinction, if the form has the attribute data-moj-add-another-enumerate="false" then the auto-enumeration should not occur. This opt-out mechanism will need to be documented.
